packages/vite/src/devFeServer.ts

```diff
@@ -112,28 +126,19 @@ async function createServer() {
   // use vite's connect instance as middleware
   app.use(vite.middlewares)
 
-  app.use('*', (req, _res, next) => {
-    const fullUrl = getFullUrl(req)
-
-    const perReqStore = createPerRequestMap({
-      // Convert express headers to fetch header
-      headers: convertExpressHeaders(req.headersDistinct),
-      fullUrl,
-    })
-
-    // By wrapping next, we ensure that all of the other handlers will use this same perReqStore
-    // But note that the serverStorage is RE-initialised for the RSC worker
-    serverStorage.run(perReqStore, next)
-  })
-
-  // Mounting middleware at /rw-rsc will strip /rw-rsc from req.url
-  app.use(
-    '/rw-rsc',
-    createRscRequestHandler({
-      getMiddlewareRouter: async () => createMiddlewareRouter(vite),
-      viteDevServer: vite,
-    }),
-  )
+  if (rscEnabled) {
+    const { createRscRequestHandler } = await import(
+      './rsc/rscRequestHandler.js'
+    )
+    // Mounting middleware at /rw-rsc will strip /rw-rsc from req.url
+    app.use(
+      '/rw-rsc',
+      createRscRequestHandler({
+        getMiddlewareRouter: async () => createMiddlewareRouter(vite),
+        viteDevServer: vite,
+      }),
+    )
+  }
```

Comment:
Are these changes related to dual mode building? If they are, how?

Reply:
Once you convert to ESM, you start getting errors like 
```
Error: The React Server Writer cannot be used outside a react-server environment. You must configure Node.js using the `--conditions react-server` flag.
```

This happens because of one (or more) of the imports in rscRequestHandler, rscWorker, etc. 

Oddly, you would think it would error out in the CJS world too, but I guess the way it builds out, it hides the issue.

Reply:
See this commit, and tests before it https://github.com/redwoodjs/redwood/pull/10934/commits/286350d84822ec060dd98d86943c4572763d36a5

Reply:
Thanks for explaining